guardrails/run.py

```diff
@@ -167,18 +177,24 @@ def prepare(
             if isinstance(prompt, Prompt):
                 prompt = prompt.format(**validated_prompt_params)
 
+            # TODO: For instructions, we're not parsing prompt params. Should we?
```

Comment:
While I can't think of a use case atm, I do think that having the flexibility to parse parameters would be handy.

Reply:
You could add a TODO for this and assign it to me. I've refactored some of the prompt formatting logic in #121 so that calling `Prompt.format(**kwargs)` only passes in the keys that are present in the prompt.

Reply:
Done